packages/datadog-instrumentations/src/mocha.js

```diff
@@ -393,8 +393,12 @@ addHook({
   file: 'lib/suite.js'
 }, (Suite) => {
   shimmer.wrap(Suite.prototype, 'addTest', addTest => function (test) {
-    // In here, the test definition is in the stack, so we search for it.
-    const startLine = getTestLineStart(new Error(), test.file)
+    const callSites = getCallSites()
+    let startLine
+    const testCallSite = callSites.find(site => site.getFileName() === test.file)
+    if (testCallSite) {
+      startLine = testCallSite.getLineNumber()
+    }
     testToStartLine.set(test, startLine)
```

Comment:
Should this line be inside the `if`? Or is it okay to call `testToStartLine.set(test, startLine)` with `startLine` being `undefined`?

Reply:
yeah it's safe. It's useless though so it'd make sense to do it inside